### DISCUSSION TITLE
Documents upload

### DISCUSSION THREAD

**Discussion by JohnnyRafael** [open-webui/open-webui#3156] [General]
# Bug Report

## Description

**Bug Summary:**
Workspace Documents Upload, I installed it on 3 different environments, one local on my MacBook and two on cloud, on debian 11 & ubuntu 24. 

I noticed an error while uploading documents to the workspace, there are files that won't get uploaded at all, there is no error message, however, all files will get uploaded into the locally hosted version. 

I created a new cloud server, thought the issue with the install, but it did the same thing. if you upload 9 documents, only 6-7 will get uploaded, rest won't and none of the missing ones will say any error. and the rejected files are random, the same file could be uploaded on the debian and won't be uploaded on the Ubuntu, and vise versa. the common file type in both situations is PDF. 

tried to upload files directly to open-webui:/app/backend/data/uploads, but none showed up on the interface. 

**Steps to Reproduce:**
navigate to workspace, documents, then upload bunch of PDFs. only few will get uploaded. 

**Expected Behavior:**
all files will get uploaded and show up or show an error message 

**Actual Behavior:**
no error message, file won't get uploaded, no status page to show whether the upload is completed or not. 

## Environment
- **Open WebUI Version:** v0.3.4
- **Ollama (if applicable):** N/A
- **Operating System:** Ubuntu 24.04 LTS & Sonoma 14.5 & Debian 11
- **Browser (if applicable):** Safari Version 17.5 & Chrome V125 

## Reproduction Details

**Confirmation:**

- [ x] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker, no Ollama, OpenAI only

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Unable to reproduce the issue here :/ Please share with us if you manage to find a fix/solution!

**Comment by JohnnyRafael** [comment]
it seems like it's issue mainly happens with the Ubuntu version, and only with some PDFs, text & Docs will get uploaded just fine. 
I re-uploaded all the missing files from the ubuntu server, to the debian one, and it worked. 

is there a way to reset everything completely ?

**Comment by JohnnyRafael** [comment]
is there a way that I can force loading the documents from the backend to the app ?

so I deleted the build and rebuilt it, uploaded all the documents again but apparently the documents are getting loaded on the backend and not the front end. 

I see all the files I uploaded inside the docker container, but I can't see them on the app. 

<img width="351" alt="Screenshot 2024-06-14 at 8 47 30 PM" src="https://github.com/open-webui/open-webui/assets/47866797/41cc7a39-8b7a-46c2-a48d-ecb969e6dd85">
<img width="257" alt="Screenshot 2024-06-14 at 8 47 50 PM" src="https://github.com/open-webui/open-webui/assets/47866797/0c7463d4-1fcb-411c-b096-7043b756c1ab">

**Comment by JohnnyRafael** [comment]
I figured out the issue, I was using the server with SSL certificate installed on it, hence I wasn't using http url, that will cause some files to drop. 

when I use the naked url, aka server ip address with http, everything works fine. 

thank you for looking into it tho. appreciated.

**Comment by saqibmahmood** [comment]
I have faces the similar situation because my instance is behind HAProxy. It is unfortunate that OpenWebUI platform doesn’t natively support SSL/TLS. Is there a way to solve this problem with the server behind the proxy for a secure connection?

**Reply by LOFWagner** [reply]
In nginx, what worked for me is adding a proxy redirect to http like this: 
 proxy_redirect http:// https://;
and also providing a client_max_body_size = 100M

**Reply by LOFWagner** [reply]
For anyone who wants to try: 
```
server {
    server_name <domain.toplevel>;  #Your IP or domain here
    client_max_body_size 100M;
    location / {
        proxy_pass http://10.0.0.2:8080;  # Your routing target here
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;


        proxy_redirect http:// https://;


        proxy_buffering off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_read_timeout 7d;

        }
```

**Reply by Skysurfer** [reply]
> For anyone who wants to try:
> 
> ```
> server {
>     server_name <domain.toplevel>;  #Your IP or domain here
>     client_max_body_size 100M;
>     location / {
>         proxy_pass http://10.0.0.2:8080;  # Your routing target here
>         proxy_set_header Host $host;
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>         proxy_set_header X-Forwarded-Proto $scheme;
> 
> 
>         proxy_redirect http:// https://;
> 
> 
>         proxy_buffering off;
>         proxy_http_version 1.1;
>         proxy_set_header Upgrade $http_upgrade;
>         proxy_set_header Connection "upgrade";
>         proxy_read_timeout 7d;
> 
>         }
> ```

Thanks for the suggestion, setting the **client_max_body_size** in the nginx configuration took care of the upload failure message for my instance.

**Reply by happyTonakai** [reply]
> In nginx, what worked for me is adding a proxy redirect to http like this: proxy_redirect http:// https://; and also providing a client_max_body_size = 100M

Your solution works for me! Without `client_max_body_size` after uploading a pdf, nothing happens.

**Comment by qubit999** [comment]
I want to add (in addition to the proposed solution above) that you can restart your nginx proxy. This solved my problem.
